app/javascript/dashboard/routes/dashboard/settings/automation/AutomationRuleRow.vue

```diff
@@ -0,0 +1,78 @@
+<script setup>
+import { messageStamp } from 'shared/helpers/timeHelper';
+
+const props = defineProps({
+  automation: {
+    type: Object,
+    required: true,
+  },
+  loading: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const emit = defineEmits(['toggle', 'edit', 'delete', 'clone']);
+
+const readableDate = date => messageStamp(new Date(date), 'LLL d, yyyy');
+const readableDateWithTime = date =>
+  messageStamp(new Date(date), 'LLL d, yyyy hh:mm a');
+
+const toggle = () => {
+  emit('toggle', {
+    id: props.automation.id,
+    name: props.automation.name,
+    status: props.automation.active,
+  });
+};
```

Comment:
Let us destructure the automation prop and use.
```suggestion
const toggle = () => {
  const { id, name, active } = props.automation;
  emit('toggle', {
    id,
    name,
    status: active,
  });
};
```